document_url/wizard/document_url.py

```diff
@@ -2,41 +2,39 @@
 # © 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
 #                       Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
 # © 2016 ACSONE SA/NV (<http://acsone.eu>)
-from openerp.osv import fields, orm
+from odoo import models, fields, api
 try:
     # Python 3
     from urllib import parse as urlparse
 except:
     from urlparse import urlparse
 
 
-class AddUrlWizard(orm.TransientModel):
+class AddUrlWizard(models.TransientModel):
     _name = 'ir.attachment.add_url'
 
-    _columns = {
-        'name': fields.char('Name', required=True),
-        'url': fields.char('URL', required=True),
-    }
+    name = fields.Char('Name', required=True)
+    url = fields.Char('URL', required=True)
 
-    def action_add_url(self, cr, uid, ids, context=None):
+    @api.multi
+    def action_add_url(self):
         """Adds the URL with the given name as an ir.attachment record."""
-        if context is None:
-            context = {}
-        if not context.get('active_model'):
+
+        context = self.env.context
+        if not context['active_model']:
             return
-        attachment_obj = self.pool['ir.attachment']
-        for form in self.browse(cr, uid, ids, context=context):
+        attachment_obj = self.env['ir.attachment']
+        for form in self:
             url = urlparse(form.url)
             if not url.scheme:
                 url = urlparse('%s%s' % ('http://', form.url))
-            for active_id in context.get('active_ids', []):
+            for active_id in context['active_ids', []]:
                 attachment = {
                     'name': form.name,
                     'type': 'url',
                     'url': url.geturl(),
-                    'user_id': uid,
                     'res_id': active_id,
-                    'res_model': context['active_model'],
+                    'res_model': context['active_model']
                 }
-                attachment_obj.create(cr, uid, attachment, context=context)
-        return {'type': 'ir.actions.act_close_wizard_and_reload_view'}
+                attachment_obj.create(attachment)
+        return False
```

Comment:
It was because in older versions of Odoo, it didn't refresh the view and you didn't see the new attachment number. I don't know if this happens also in v10.